src/pgduckdb_duckdb.cpp

```diff
@@ -230,6 +224,21 @@ DuckDBManager::CreateConnection() {
 	    duckdb::StringUtil::Format("SET http_file_cache_dir TO '%s';", CreateOrGetDirectoryPath("duckdb_cache"));
 	context.Query(http_file_cache_set_dir_query, false);
 
+	if (!superuser() && duckdb_disabled_filesystems != NULL) {
```

Comment:
nit- looks like `superuser()` can be non trivial, so:
```suggestion
	if (duckdb_disabled_filesystems != NULL && !superuser()) {
```